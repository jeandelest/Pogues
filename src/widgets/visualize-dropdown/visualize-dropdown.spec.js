import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import VisualizeDropdown from './components/visualize-dropdown';

describe('Visualize Dropdown Component: ', () => {
  test('Should return the right HTML', () => {
    const props = {
      visualizeActiveQuestionnaire() {},
      disabled: false,
      top: false,
      componentId: 'component-id',
    };
    const tree = renderer.create(<VisualizeDropdown {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('Should display the dropdown on top', () => {
    const props = {
      visualizeActiveQuestionnaire() {},
      disabled: false,
      top: true,
      componentId: 'component-id',
    };
    const tree = renderer.create(<VisualizeDropdown {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('Should disable the button', () => {
    const props = {
      visualizeActiveQuestionnaire() {},
      disabled: true,
      top: false,
      componentId: 'component-id',
    };
    const tree = renderer.create(<VisualizeDropdown {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('Should toggle the dropdown', () => {
    const props = {
      visualizeActiveQuestionnaire() {},
      disabled: true,
      top: false,
      componentId: 'component-id',
    };
    const wrapper = shallow(<VisualizeDropdown {...props} />);
    expect(wrapper.find('div').hasClass('open')).toBeFalsy();
    wrapper
      .find('button[id="visualize"]')
      .simulate('click', { preventDefault() {}, stopPropagation() {} });
    expect(wrapper.find('div').hasClass('open')).toBeTruthy();
    wrapper
      .find('button[id="visualize"]')
      .simulate('click', { preventDefault() {}, stopPropagation() {} });
    expect(wrapper.find('div').hasClass('open')).toBeFalsy();
  });

  test('Should call the visualizeActiveQuestionnaire method and hide the dropdown', () => {
    const props = {
      visualizeActiveQuestionnaire: jest.fn(),
      disabled: true,
      top: false,
      componentId: 'component-id',
    };
    const wrapper = shallow(<VisualizeDropdown {...props} />);
    wrapper
      .find('a')
      .first()
      .simulate('click', { preventDefault() {}, stopPropagation() {} });
    expect(props.visualizeActiveQuestionnaire).toHaveBeenCalledWith(
      'html',
      props.componentId,
      props.token,
    );
  });
});
