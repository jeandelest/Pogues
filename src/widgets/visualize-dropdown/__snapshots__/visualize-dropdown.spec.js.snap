// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualize Dropdown Component:  Should disable the button 1`] = `
<div
  className="btn-group flex-column"
>
  <button
    aria-expanded={false}
    aria-haspopup="true"
    className="btn dropdown-toggle btn-yellow disabled"
    data-toggle="dropdown"
    disabled={true}
    id="visualize"
    onClick={[Function]}
  >
    Visualise
    <span
      className="caret"
    />
  </button>
  <ul
    className="dropdown-menu"
  >
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V1
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V2
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V3
        <span
          className="link-tag"
        >
          <i>
            beta
          </i>
        </span>
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Interviewer face to face
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Interviewer phone
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Paper
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Specification
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Metadata
      </a>
    </li>
  </ul>
</div>
`;

exports[`Visualize Dropdown Component:  Should display the dropdown on top 1`] = `
<div
  className="btn-group dropup flex-column-reverse"
>
  <button
    aria-expanded={false}
    aria-haspopup="true"
    className="btn dropdown-toggle btn-yellow"
    data-toggle="dropdown"
    disabled={false}
    id="visualize"
    onClick={[Function]}
  >
    Visualise
    <span
      className="caret"
    />
  </button>
  <ul
    className="dropdown-menu"
  >
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V1
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V2
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V3
        <span
          className="link-tag"
        >
          <i>
            beta
          </i>
        </span>
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Interviewer face to face
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Interviewer phone
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Paper
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Specification
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Metadata
      </a>
    </li>
  </ul>
</div>
`;

exports[`Visualize Dropdown Component:  Should return the right HTML 1`] = `
<div
  className="btn-group flex-column"
>
  <button
    aria-expanded={false}
    aria-haspopup="true"
    className="btn dropdown-toggle btn-yellow"
    data-toggle="dropdown"
    disabled={false}
    id="visualize"
    onClick={[Function]}
  >
    Visualise
    <span
      className="caret"
    />
  </button>
  <ul
    className="dropdown-menu"
  >
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V1
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V2
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Web V3
        <span
          className="link-tag"
        >
          <i>
            beta
          </i>
        </span>
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Interviewer face to face
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Interviewer phone
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Paper
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Specification
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        Metadata
      </a>
    </li>
  </ul>
</div>
`;
